crowbar_framework/app/models/service_object.rb

```diff
@@ -1271,6 +1271,8 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     pre_cached_nodes = {}
 
     # Each batch is a list of nodes that can be done in parallel.
+    # only run remote chef-clent on a node once
+    batches.uniq!
```

Comment:
AFAICS ```batches``` is a list of a lists. So I don't think this really does what you want it to do. E.g. assuming you have a setup of two nodes where node1 is member of ```provisioner-server``` and ```provisioner-base``` and node2 just ```provisioner-base```. Won't batches then look like this: ```[["node1"], ["node1", "node2"]]```? Calling ```uniq!``` on that list does not have any effect.